.github/workflows/test.yml

```
@@ -7,7 +7,30 @@ on:
   workflow_dispatch:
   workflow_call:
 jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
```

You need a valid python version here, or it will break actions/setup-python@v4's caching mechanism. 
```suggestion
          python-version: "3.11"
```